tests/test_editable_generated.py

```diff
@@ -0,0 +1,304 @@
+"""Test regular and editable installs with generated files.
+
+Illustrate the supported and correct ways to use generated files
+(other than traditional compiled extension modules).
+
+Check a variety of scenarios in which package files (modules or data) are
+not present in the source tree to confirm that we can find resources as expected,
+either by ``import`` or with tools such as `importlib.resources.files()`.
+"""
+
+from __future__ import annotations
+
+import sys
+from pathlib import Path
+
+import pytest
+from conftest import PackageInfo, VEnv, process_package
+
+
+def _setup_package_for_editable_layout_tests(
+    monkeypatch: pytest.MonkeyPatch,
+    tmp_path: Path,
+    editable: bool,
+    editable_mode: str,
+    build_isolation: bool,
+    isolated: VEnv,
+) -> None:
+    editable_flag = ["-e"] if editable else []
+
+    config_mode_flags = []
+    if editable:
+        config_mode_flags.append(f"--config-settings=editable.mode={editable_mode}")
+    if editable_mode != "inplace":
+        config_mode_flags.append("--config-settings=build-dir=build/{wheel_tag}")
+
+    build_isolation_flags = []
+    if not build_isolation:
+        build_isolation_flags.append("--no-build-isolation")
+
+    # Use a context so that we only change into the directory up until the point where
+    # we run the editable install. We do not want to be in that directory when importing
+    # to avoid importing the source directory instead of the installed package.
+    with monkeypatch.context() as m:
+        package = PackageInfo("cmake_generated")
+        process_package(package, tmp_path, m)
+
+        assert isolated.wheelhouse
+
+        ninja = [
+            "ninja"
+            for f in isolated.wheelhouse.iterdir()
+            if f.name.startswith("ninja-")
+        ]
+        cmake = [
+            "cmake"
+            for f in isolated.wheelhouse.iterdir()
+            if f.name.startswith("cmake-")
+        ]
+
+        isolated.install("pip>23")
+        isolated.install("scikit-build-core", *ninja, *cmake)
```

Comment:
We only need this for `not build_isolation` case. Would also be nice if this was generalized since this is quite common in other parts of the code also. Maybe as a method for `Venv`

Reply:
addressed in suggested change(?)